### PR TITLE
making error msg for malformed packages.yml better

### LINKED ISSUE #2017: Unhelpful error message with malformed packages.yml
### Describe the bug
```
MacBook-Pro:analytics tristan$ cat packages.yml
packages:
  - package: fishtown-analytics/stripe
    revision: 0.3.1
```

I had recently switched from a git reference to this package to a hub reference, and forgot that I needed to change from `revision` to `version`. This is the error message I received, which made it significantly more challenging to debug:

```
MacBook-Pro:analytics tristan$ dbt clean && dbt deps
Running with dbt=0.15.0
Encountered an error while reading the project:
  ERROR: Runtime Error
  Invalid package config: at path []: 'local' is a required property
```

### Expected behavior
I would expect an error message like:
```
Required parameter version not supplied.
```
...or something similar.

**The output of `dbt --version`:**
```
MacBook-Pro:analytics tristan$ dbt --version
installed version: 0.15.0
   latest version: 0.15.0

Up to date!
```

### PR DESCRIPTION
**Pull request by sonac**
I messed up the [previous](https://github.com/fishtown-analytics/dbt/pull/2077) PR during rebase and it was automatically closed. Here's new and clean one.

Fixes #2017

### CODE REVIEW

File: core/dbt/config/project.py [hunk 1]
```diff
@@ -59,6 +59,11 @@
 --no-version-check
 """
 
+MALFORMED_PACKAGE_ERROR = """\
+packages.yml is malformed, please make sure that it matches
```
Comment (drewbanin) [inline core/dbt/config/project.py]: Check out these copy suggestions. Couple of things:
1. We like using full sentences in these error messages
2. I picked a more generic link so we don't accidentally break this in the docs in the future
3. We should retain the error message (see below) in this error output

```suggestion
The packages.yml file in this project is malformed. Please double check the contents
of this file and fix any errors before retrying.

You can find more information on the syntax for this file here:
https://docs.getdbt.com/docs/package-management

Validator Error:
{error}
```

File: core/dbt/config/project.py [hunk 2]
```diff
@@ -130,7 +135,7 @@ def package_config_from_data(packages_data):
         packages = PackageConfig.from_dict(packages_data)
     except ValidationError as e:
         raise DbtProjectError(
-            'Invalid package config: {}'.format(validator_error_message(e))
+            MALFORMED_PACKAGE_ERROR
```
Comment (drewbanin) [inline core/dbt/config/project.py]: Let's retain the error message here. It is helpful, and it definitely benefits from some additional context. I think the stringified error message is a little bit easier to understand than the output of `validator_error_message()` here though

```suggestion
            MALFORMED_PACKAGE_ERROR.format(error=str(e))
```

File: core/dbt/config/project.py (hunk 1 above)
Comment (sonac) [reply core/dbt/config/project.py]: noted, though regarding validation error retainment - the initial issue was raised that it is misleading, so i was doubtful regarding including it into the message

File: core/dbt/config/project.py [hunk 3]
```diff
@@ -135,7 +142,7 @@ def package_config_from_data(packages_data):
         packages = PackageConfig.from_dict(packages_data)
     except ValidationError as e:
         raise DbtProjectError(
-            MALFORMED_PACKAGE_ERROR
+            MALFORMED_PACKAGE_ERROR.format(error=str(e))
```
Comment (drewbanin) [inline core/dbt/config/project.py]: Ooops - I think I led you astray here - we're actually going to want `str(e.message)`

```suggestion
            MALFORMED_PACKAGE_ERROR.format(error=str(e.msg))
```

In practice, this should result in output like:

Screen Shot 2020-01-31 at 10 45 06 AM

Which i think is pretty good all things considered!

Note: we'll also want to remove the duplicated link on line 71!

Comment (sonac) [reply core/dbt/config/project.py]: Thanks for the review!

File: core/dbt/config/project.py [hunk 4]
```diff
@@ -59,6 +59,17 @@
 --no-version-check
 """
 
+MALFORMED_PACKAGE_ERROR = """\
+The packages.yml file in this project is malformed. Please double check the contents
```
Comment (drewbanin) [inline core/dbt/config/project.py]: I see that our style guide checker (flake8) failed with this error:

> core/dbt/config/project.py:63:80: E501 line too long (84 > 79 characters)

Can you just split this onto two lines to appease flake8? Thanks!

File: core/dbt/config/project.py [hunk 5]
```diff
@@ -59,6 +59,17 @@
 --no-version-check
 """
 
+MALFORMED_PACKAGE_ERROR = """\
+The packages.yml file in this project is malformed. Please double check
```
Comment (drewbanin) [inline core/dbt/config/project.py]:
```
flake8 run-test: commands[0] | /bin/bash -c '$(which flake8) --select=E,W,F --ignore=W504 core/dbt plugins/*/dbt'
core/dbt/config/project.py:63:72: W291 trailing whitespace
```

```suggestion
The packages.yml file in this project is malformed. Please double check
```

**Comment by drewbanin** [comment]
Thanks for sticking with this one @sonac! Merging this for our 0.16.0 release :D

**Pull request merged**